splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -975,25 +1137,46 @@ public V getThird(){
 
 
     public static void SYSCS_GET_GLOBAL_DATABASE_PROPERTY(final String key,final ResultSet[] resultSet) throws SQLException{
-        Map<String,String> valueMap = EngineDriver.driver().dbAdministrator().getGlobalDatabaseProperty(key);
 
-        ResultSetBuilder rsBuilder=new ResultSetBuilder();
-        try{
-            rsBuilder.getColumnBuilder()
-                    .addColumn("HOST_NAME",Types.VARCHAR,32)
-                    .addColumn("PROPERTY_VALUE",Types.VARCHAR,128);
-            RowBuilder rowBuilder=rsBuilder.getRowBuilder();
-            int hostIdx=0;
-            for(Map.Entry<String,String> entry : valueMap.entrySet()){
-                rowBuilder.getDvd(0).setValue(entry.getKey());
-                rowBuilder.getDvd(1).setValue(entry.getValue());
-                rowBuilder.addRow();
-                hostIdx++;
+        List<HostAndPort> servers;
+        try {
+            servers = EngineDriver.driver().getServiceDiscovery().listServers();
+        } catch (IOException e) {
+            throw PublicAPI.wrapStandardException(Exceptions.parseException(e));
+        }
+
+        List<ExecRow> rows = new ArrayList<>();
+        for (HostAndPort server : servers) {
+            try (Connection connection = RemoteUser.getConnection(server.toString())) {
+                try (PreparedStatement ps = connection.prepareStatement("values SYSCS_UTIL.SYSCS_GET_DATABASE_PROPERTY(?)")) {
+                    ps.setString(1, key);
+                    try (ResultSet rs = ps.executeQuery()) {
+                        while (rs.next()) {
+                            ExecRow row = new ValueRow(1);
```

Comment:
ValueRow should be initialized with 2 fields instead of 1.

Reply:
Good catch, thanks!